Neos.Neos/Classes/Domain/Model/UserId.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Model;
+
+/**
+ * Globally unique identifier of a Neos user
+ *
+ * @api
+ */
+final readonly class UserId implements \JsonSerializable
```

Comment:
will this replace the cr's user id thing?

Reply:
It probably won't _replace_ it because the CR also has a concept of users (currently only used to store the `initiatingUserId` in the event metadata).
But IMO it makes sense to have a dedicated "Neos UserId" as well.. Maybe we can rename one to make the distinction clearer..